ff/serving/metadata/metadata.go

```diff
@@ -801,11 +806,37 @@ func NewMetadataServer(server *Config) (*MetadataServer, error) {
 		}
 	}
 	return &MetadataServer{
-		lookup: lookup,
-		Logger: server.Logger,
+		lookup:  lookup,
+		address: config.Address,
+		Logger:  config.Logger,
 	}, nil
 }
 
+func (serv *MetadataServer) Serve() error {
+	if serv.grpcServer != nil {
+		return fmt.Errorf("Server already running")
+	}
+	lis, err := net.Listen("tcp", serv.address)
+	if err != nil {
+		return err
+	}
+	grpcServer := grpc.NewServer()
+	pb.RegisterMetadataServer(grpcServer, serv)
+	serv.grpcServer = grpcServer
+	serv.Logger.Infow("Server starting", "Address", serv.address)
+	return grpcServer.Serve(lis)
+}
+
+func (serv *MetadataServer) GracefulStop() {
+	serv.grpcServer.GracefulStop()
```

Comment:
Does this throw an error if there is no grpc server currently running?

Reply:
No, but the internally grpcServer doesn't either. I just mimicked it's behavior